### DISCUSSION TITLE
Question: The best way to interact with Vue v-model?

### DISCUSSION THREAD

**Discussion by PurpleTape** [uNmAnNeR/imaskjs#893] [Q&A]
Здравствуйте!

Подскажите, пожалуйста, как лучше обработать такой случай с Vue, когда состояние `input` хранится в реактивном значении и реактивное значение является единственным источником истины для `input`? Пример:
```vue
<template>
  <input v-model="value"/>
</template>

<script setup>
  import type { Ref } from 'vue';

  const value = ref('');
</script>
```
IMask добавляет собственные обработчики событий (`onInput`, `onKeydown` и тд) и ориентируется на значение `input.value`. Таким образом, при ситуации, когда значение `input` не будет соответствовать заданной маске IMask, то обработчики событий не обновят внутреннее значение (`masked.value` и `masked.unmaskedValue`), но Vue обновит значение `v-model` и произойдет рассинхронизация значений `input.value`, `masked.value`, а затем возникнет сообщение "Element value was changed outside of mask. Syncronize mask using mask.updateValue() to work properly".

При этом значение `v-model` обновляет содержимое `input` не в момент установки (как `input.value = 'new value'`), а в следующем тике (группой, вместе с другими обновлениями)

Также хочу выразить благодарность за проделанную работу над библиотекой!

Спасибо!

**Comment by uNmAnNeR** [comment]
Привет! а плагин для ВЬЮ не пробовали? вроде там все это обрабатывается. Есть директива, компонент или хук на выбор.

**Comment by roma-marshall** [comment]
Take a look at this: https://vuejs.org/guide/components/v-model.html
